commons/src/main/java/io/aiven/kafka/tieredstorage/commons/io/SegmentCompressionChecker.java

```diff
@@ -0,0 +1,28 @@
+package io.aiven.kafka.tieredstorage.commons.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Function;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+
+/**
+ * Checks if segment requires compression or not.
+ * To be used when segment files are received on archival.
+ */
+public class SegmentCompressionChecker implements Function<File, Boolean> {
+
+    /**
+     * @param file Kafka log segment file
+     * @return true if requires compression, else false
+     */
+    @Override
+    public Boolean apply(File file) {
+        try (final FileRecords records = FileRecords.open(file)){
+            return records.firstBatch().compressionType() == CompressionType.NONE;
+        } catch (IOException e) {
```

Comment:
Just to check: what does Kafka throw on various validation errors, e.g. CRC mismatch?

Reply:
Good question. For safety we may extend the catch to Exception, and make a note on it, wdyt?

Reply:
Yeah, makes sense. Probably also some logging wouldn't be bad.

Reply:
Sure, adding logging.